com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -35,68 +35,96 @@ public enum BehaviorType
     /// The Factory to generate policies.
     /// </summary>
     [AddComponentMenu("ML Agents/Behavior Parameters", (int)MenuGroup.Default)]
-    internal class BehaviorParameters : MonoBehaviour
+    public class BehaviorParameters : MonoBehaviour
     {
-        [HideInInspector]
-        [SerializeField]
+        [HideInInspector, SerializeField]
         BrainParameters m_BrainParameters = new BrainParameters();
-        [HideInInspector]
-        [SerializeField]
-        NNModel m_Model;
-        [HideInInspector]
-        [SerializeField]
-        InferenceDevice m_InferenceDevice;
-        [HideInInspector]
-        [SerializeField]
 
-        // Disable warning /com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs(...):
-        //   warning CS0649: Field 'BehaviorParameters.m_BehaviorType' is never assigned to,
-        //   and will always have its default value
-        // This field is set in the custom editor.
-        #pragma warning disable 0649
-        internal BehaviorType m_BehaviorType;
-        #pragma warning restore 0649
-        [HideInInspector]
-        [SerializeField]
-        string m_BehaviorName = "My Behavior";
+        /// <summary>
+        /// The associated <see cref="BrainParameters"/> for this behavior.
+        /// </summary>
+        public BrainParameters brainParameters
+        {
+            get { return m_BrainParameters; }
+            internal set { m_BrainParameters = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        NNModel m_Model;
 
         /// <summary>
-        /// The team ID for this behavior.
+        /// The neural network model used when in inference mode.
+        /// This cannot be set directly; use <see cref="Agent.GiveModel(string,NNModel,InferenceDevice)"/>
```

Comment:
I think this is the right approach; we should all direct `set` on this, the model, or the inference device.